deps/utf8proc.mk

```diff
@@ -1,34 +1,40 @@
 ## UTF8PROC ##
 UTF8PROC_GIT_URL := git://github.com/JuliaLang/utf8proc.git
 UTF8PROC_TAR_URL = https://api.github.com/repos/JuliaLang/utf8proc/tarball/$1
-$(eval $(call git-external,utf8proc,UTF8PROC,Makefile,libutf8proc.a,$(BUILDDIR)))
+$(eval $(call git-external,utf8proc,UTF8PROC,,,$(BUILDDIR)))
 
-UTF8PROC_SRC_TARGET := $(BUILDDIR)/$(UTF8PROC_SRC_DIR)/libutf8proc.a
 UTF8PROC_OBJ_LIB    := $(build_libdir)/libutf8proc.a
 UTF8PROC_OBJ_HEADER := $(build_includedir)/utf8proc.h
-UTF8PROC_OBJ_TARGET := $(UTF8PROC_OBJ_LIB) $(UTF8PROC_OBJ_HEADER)
 UTF8PROC_CFLAGS     := -O2
 UTF8PROC_MFLAGS     := CC="$(CC) $(DEPS_CFLAGS)" CFLAGS="$(CFLAGS) $(UTF8PROC_CFLAGS)" PICFLAG="$(fPIC)" AR="$(AR)"
 
-$(UTF8PROC_SRC_TARGET): $(BUILDDIR)/$(UTF8PROC_SRC_DIR)/Makefile
+$(BUILDDIR)/$(UTF8PROC_SRC_DIR)/build-compiled: $(BUILDDIR)/$(UTF8PROC_SRC_DIR)/source-extracted
 	$(MAKE) -C $(dir $<) $(UTF8PROC_MFLAGS) libutf8proc.a
-	touch -c $@
-$(BUILDDIR)/$(UTF8PROC_SRC_DIR)/checked: $(UTF8PROC_SRC_TARGET)
+	echo 1 > $@
+
+$(BUILDDIR)/$(UTF8PROC_SRC_DIR)/build-checked: $(BUILDDIR)/$(UTF8PROC_SRC_DIR)/build-compiled
 ifeq ($(OS),$(BUILD_OS))
 	$(MAKE) -C $(dir $@) $(UTF8PROC_MFLAGS) check
 endif
 	echo 1 > $@
-$(UTF8PROC_OBJ_LIB): $(UTF8PROC_SRC_TARGET)
-	cp -f $< $@
-$(UTF8PROC_OBJ_HEADER): $(UTF8PROC_SRC_TARGET)
-	cp -f $(BUILDDIR)/$(UTF8PROC_SRC_DIR)/utf8proc.h $@
+
+define UTF8PROC_INSTALL
+	mkdir -p $2/$$(build_includedir) $2/$$(build_libdir)
+	cp $1/utf8proc.h $2/$$(build_includedir)
+	cp $1/libutf8proc.a $2/$$(build_libdir)
+endef
+$(eval $(call staged-install, \
+	utf8proc,$(UTF8PROC_SRC_DIR), \
+	UTF8PROC_INSTALL,,,))
 
 clean-utf8proc:
+	-rm $(BUILDDIR)/$(UTF8PROC_SRC_DIR)/build-compiled
 	-$(MAKE) -C $(BUILDDIR)/$(UTF8PROC_SRC_DIR) clean
-	-rm -rf $(build_libdir)/libutf8proc.a $(build_includedir)/utf8proc.h
 
 get-utf8proc: $(UTF8PROC_SRC_FILE)
-configure-utf8proc: $(BUILDDIR)/$(UTF8PROC_SRC_DIR)/Makefile
-compile-utf8proc: $(UTF8PROC_SRC_TARGET)
-check-utf8proc: $(BUILDDIR)/$(UTF8PROC_SRC_DIR)/checked
-install-utf8proc: $(UTF8PROC_OBJ_TARGET)
+extract-utf8proc: $(BUILDDIR)/$(UTF8PROC_SRC_DIR)/source-extracted
+configure-utf8proc: extract-utf8proc
+compile-utf8proc: $(BUILDDIR)/$(UTF8PROC_SRC_DIR)/build-compiled
+# utf8proc tests disabled since they require a download
```

Comment:
the only buildbots that don't have network access are the ones that use system versions of deps for nearly everything